dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,718 @@
+focused on RISC-V, which right now is the only architecture supported. However,
+x86 already contains
+[dummy implementation](https://github.com/betrusted-io/xous-core/blob/30b82b25b100e958790973c129dc49e1acca79ec/kernel/src/arch/x86_64.rs)
+which probably will be extended someday.
+
+
+#### Running in DLME
+
+Xous currently cannot run on x86, and DLME on RISC-V is not covered here. We
+have opened an [issue](https://github.com/betrusted-io/xous-core/issues/139)
+raising the subject, among others, about x86 support.
+
+#### Fobnail integration
+
+Xous has network support but currently lacks USB host drivers and TPM support.
+Also, it lacks C interface for calling the kernel and lacks C library. This
+should not be a serious problem and may be solved by rewriting
+[Fobnail Attester](https://github.com/fobnail/fobnail-attester) in Rust or by
+adding required C bindings to Rust code. It means that getting
+[Fobnail Attester](https://github.com/fobnail/fobnail-attester) running may
+require some amount of work. Also, Xous needs to gain kexec-like abilities to
+chainload target OS.
+
+### seL4
+
+seL4 is a secure L4 family microkernel written in C. It has strong security
+guarantees assured by [formal proofs](https://sel4.systems/Info/FAQ/proof.pml).
+However these proofs are still incomplete for x86; see
+[Supported Platforms](https://docs.sel4.systems/Hardware) for an up-to-date
+verification status. Due to its microkernel nature, seL4 provides stronger
+isolation. A breach in one of the components (such as a USB driver or the
+network stack) wouldn't compromise the entire OS (contrary to monolithic
+kernels).
+
+We have opened an [issue](https://github.com/seL4/seL4/issues/832) about seL4
+usage as secure bootloader.
+
+#### Running in DLME
+
+It should be runnable, however due to microkernel nature all programs run in
+unprivileged mode which complicates booting of the target OS. seL4 would have to
+gain kexec-like abilities. seL4 developers are
+[willing](https://github.com/seL4/seL4/issues/832#issuecomment-1114385689) to
+add kexec support.
+
+#### Fobnail integration
+
+seL4 provides virtually no drivers, except a few drivers listed
+[here](https://docs.sel4.systems/projects/available-user-components.html#device-drivers).
+According to the page linked above, `libusbdrivers` is inactive and lacks XHCI
+support. seL4 has basic support for an old version of musl C library (v1.1.16).
+
+Using seL4 would require a significant amount of work
+- Extending USB drivers
+- Implementing USB EEM driver
+- Implementing TPM driver
+- Porting [Fobnail Attester](https://github.com/fobnail/fobnail-attester)
+- seL4 runs all processes in unprivileged mode, so the kernel itself would have
+  to be modified to allow booting of another kernel
+
+[Genode](https://github.com/genodelabs/genode) can provide many of the required
+drivers. See the section below for details.
+
+CAmkES provides tools for writing seL4-native components, it also has VMM
```

Comment:
What is CAmkES, some link also would be useful.

Reply:
Link added on the first CAmkES appearance